src/Result.php

```diff
@@ -242,4 +243,52 @@ private function ensureHasKeyValue(): void
             throw NoKeyValue::fromColumnCount($columnCount);
         }
     }
+
+    /**
+     * BC layer for a wide-spread use-case of old DBAL APIs
+     *
+     * @deprecated This API is deprecated and will be removed after 2022
+     *
+     * @return array<array-key,mixed>|false
+     */
+    public function fetch(int $mode = FetchMode::ASSOCIATIVE)
+    {
+        if ($mode === FetchMode::ASSOCIATIVE) { // PDO::FETCH_ASSOC
+            return $this->fetchAssociative();
+        }
+
+        if ($mode === FetchMode::NUMERIC) { // PDO::FETCH_NUM
+            return $this->fetchNumeric();
+        }
+
+        if ($mode === FetchMode::COLUMN) { // PDO::FETCH_COLUMN
+            return $this->fetchOne();
+        }
+
+        throw new RuntimeException('only fetch mode PDO::FETCH_ASSOC and PDO::FETCH_NUM are supported by legacy API.');
```

Comment:
Is the exception message starting with a lower-case letter on purpose? You must have written some Golang recently :-)